docs/guides/jupyter.rst

```diff
@@ -0,0 +1,293 @@
+- In Visual Studio Code `it's not possible to save the widget
+  state <https://github.com/microsoft/vscode-jupyter/issues/4404>`_
+  at the time of writing.
+
+.. _ipywidgets: https://ipywidgets.readthedocs.io/
+
+.. figure:: /_static/images/guides/jupyterlab-save-widget-state.png
+   :width: 30%
+   :align: center
+   :alt: JupyterLab option to save the interactive widget state automatically
+
+   JupyterLab option to save the interactive widget state automatically
+
+For example, if you create a notebook with a simple
+`IntSlider <https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html#IntSlider>`_
+widget from ipywidgets and save the widget state,
+the slider will render correctly in Sphinx.
+
+.. figure:: /_static/images/guides/widget-html.gif
+   :width: 80%
+   :align: center
+   :alt: Interactive widget rendered in HTML by Sphinx
+
+   Interactive widget rendered in HTML by Sphinx
+
+To see more elaborate examples:
+
+- `ipyleaflet`_ provides several widgets for interactive maps,
+  and renders live versions of them `in their
+  documentation <https://ipyleaflet.readthedocs.io/en/latest/api_reference/velocity.html>`_.
+- `PyVista <https://docs.pyvista.org/>`_ is used for scientific 3D visualization
+  with several interactive backends and `examples in their
+  documentation <https://docs.pyvista.org/index.html#maps-and-geoscience>`_ as well.
+
+.. warning::
+
+   Although widgets themselves can be embedded in HTML as shown above,
+   `events <https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20Events.html>`_
+   require a backend (kernel) to execute.
+   Therefore, ``@interact``, ``.observe``, and related functionalities relying on them
+   will not work as expected.
+
+Creating galleries of examples using notebooks
+----------------------------------------------
+
+If you want to create an HTML gallery of examples for your project,
+`Sphinx-Gallery <https://sphinx-gallery.github.io/>`_
+offers a very convenient way of doing so from Sphinx,
+including thumbnail generation.
+However, `it works with Python scripts rather than
+notebooks <https://github.com/sphinx-gallery/sphinx-gallery/issues/245>`_,
+which might not be what you want.
+
+`nbsphinx`_ has support for `creating thumbnail galleries from a list of Jupyter
+notebooks <https://nbsphinx.readthedocs.io/en/latest/subdir/gallery.html>`_,
+and it is compatible with Sphinx-Gallery styles.
+To use it, you will need to install both nbsphinx and Sphinx-Gallery,
+and modify your ``conf.py`` as follows:
+
+.. code-block:: python
+   :caption: conf.py
+
+   extensions = [
+      'nbsphinx',
+      'sphinx_gallery.load_style',
+   ]
+
+After doing that, there are two ways to create the gallery:
+
+- From a reStructuredText source file, using the ``.. nbgallery::`` directive,
+  `as showcased in the
+  documentation <https://nbsphinx.readthedocs.io/en/latest/a-normal-rst-file.html#thumbnail-galleries>`_.
```

Comment:
Should we use intersphinx here?

Reply:
If only I had something like https://webknjaz.github.io/intersphinx-untangled/ to quickly inspect the inventory of a project, to see if this would make sense or not...!

I don't see any labels defined in https://github.com/spatialaudio/nbsphinx/blame/master/doc/a-normal-rst-file.rst, so I suppose we can't? But I might be missing some information here. cc @humitos

Reply:
@astrojuanlu isn't that just a web-based version of the tool built into intersphinx? https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#showing-all-links-of-an-intersphinx-mapping-file

Reply:
```
	/a-normal-rst-file.rst#thumbnail-galleries Thumbnail Galleries                     : a-normal-rst-file.html#thumbnail-galleries
```

Reply:
TIL about `python -msphinx.ext.intersphinx` 👀 looking into this

Reply:
Right, I had a misconception about how `:ref:` worked that had nothing to do with the inventory. Basically I didn't know I could do this:

```
:ref:`as showcased in the documentation <nbsphinx:/a-normal-rst-file.rst#thumbnail-galleries>`
```

Reviewing the whole guide now. Thanks @ericholscher !

Reply:
Done. I converted some of the references to intersphinx, with these criteria:

- All the "core" references, except for a couple of Sphinx references I didn't know how to link.
- Examples for "social proof" (poliastro, ipyleaflet) _not_ converted, to avoid overloading the intersphinx mapping (not sure if I should worry about this?)